packages/hooks/src/use-data-table-sorting-state/use-data-table-sorting-state.ts

```diff
@@ -0,0 +1,51 @@
+import { useState, useCallback } from 'react';
+import isNil from 'lodash/isNil';
+
+type TSortDefinition = {
+  key: string;
+  order: 'desc' | 'asc';
+};
+
+type TDataTableSortingState = {
+  value: TSortDefinition;
+  onChange: (
+    key: TSortDefinition['key'],
+    order: TSortDefinition['order']
```

Comment:
I was tempted to update the callback signature but that would be a breaking change. I played around with Types and thought maybe this is a great use case to use function overloads and let the type describe intent with the signature

https://gist.github.com/adnasa/66611e1fb9064074caa2eaf17297562e